lib/rules/lint-no-childless-elements.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const Rule = require('./base');
+
+const BLOCK_TAGS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'span', 'a', 'ul', 'ol', 'li', 'div'];
```

Comment:
Where exactly does this list come from? Is it actually complete?

Would it be easier to flip this around and specify the void elements here instead? I see the list `area, base, br, col, embed, hr, img, input, link, meta, param, source, track, wbr` from https://www.w3.org/TR/html/syntax.html#void-elements

Reply:
These are the block level elements in HTML:
`<address><article><aside><blockquote><canvas><dd><div><dl><dt><fieldset><figcaption><figure><footer><form><h1>-<h6><header><hr><li><main><nav><noscript><ol><p><pre><section><table><tfoot><ul><video>`

Reply:
Resolving discussion, as list and logic behind evolved